crypto/evp/p_lib.c

```diff
@@ -269,6 +269,10 @@ static int evp_pkey_cmp_any(const EVP_PKEY *a, const EVP_PKEY *b,
     if (keymgmt1 != keymgmt2)
         return -2;
 
+    /* If the keymgmt implementations are NULL, the export failed */
```

Comment:
If keymgmt1 == NULL does this imply that a->keymgmt is also NULL??
I notice there is call
```
EVP_KEYMGMT_is_a(a->keymgmt...
```
near the top of this function.
Which then dereferences the (NULL?) pointer internally.

Reply:
No.  Read the comment.  It means that one of the `evp_pkey_export_to_provider() calls a few lines up failed.

Reply:
This is incorrect, while doing the explicit ec work - I managed to get a crash in evp_keymgmt_match() because keymgmt1 was NULL. i.e- the export process failed.

Reply:
Er, you seem to confirm what I say. How's that incorrect?

Reply:
Ahh yeah.. Iwas looking at code without the change :)